components/QuizDialog.tsx

```diff
@@ -40,7 +40,7 @@ export default function QuizDialog({
   useEffect(() => {
     if (open) {
       const shuffled = [...cards].sort(() => Math.random() - 0.5);
-      setQuizCards(shuffled.slice(0, Math.min(5, cards.length)));
+      setQuizCards(shuffled.slice(0, 5));
```

Comment:
index bounds checking
you can request a quiz with more questions than there are cards in the deck. in this case the quiz will work, but only show as many questions as you have added to the deck. additionally, if you requested 5 questions and your deck only has 3 cards, if you get them all right your score will show 3/5.